### ISSUE TITLE
Unable to install C extension on PHP 5.5.20

### ISSUE THREAD

**Issue by tonglil** [lstrojny/functional-php#47]
Both `sudo pecl install functional-0.6.0` and `sudo pear upgrade -f pecl/functional` end with `ERROR: `make' failed`.

Based on the stack trace, I think it's incompatabilities with the new PHP ZendEngine. The log filtering for 'error':

```
/tmp/pear/temp/functional/functional.c:451:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:451:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:492:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:492:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:539:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:539:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:585:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:585:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:629:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:629:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:675:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:675:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:719:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:719:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:777:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:777:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:830:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:830:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:880:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:880:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:941:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:941:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:999:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:999:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1045:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1045:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1098:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1098:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1153:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1153:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1200:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1200:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1333:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1333:4: error: void value not ignored as it ought to be
/tmp/pear/temp/functional/functional.c:1554:4: error: too many arguments to function ‘zend_user_it_get_current_key’
/tmp/pear/temp/functional/functional.c:1554:4: error: void value not ignored as it ought to be
```

I can post the full logs if you want.

**Comment by lstrojny** [comment]
Full logs would be appreciated. It’s really weird since travis tests are working: https://travis-ci.org/lstrojny/functional-php/jobs/46633089. Could you try installing master (not much change in the C extension anyway).

**Comment by tonglil** [comment]
I was able to install using phpize, but how can I try using it?

```php
<?php
$a = ['one' => 1, 'two' => 2, 'three' => 3];
F::pick($a, 'one');
F\pick($a, 'one');
```

**Comment by tonglil** [comment]
And here are the full traces for the pear/pecl installs: https://gist.github.com/tonglil/c8cc3432e60525e583aa

**Comment by lstrojny** [comment]
Won’t fix as the C extension will be removed in 1.2
